### PR TITLE
Helmet tags attributes can be presented in any order

### LINKED ISSUE #80: Server Meta ignored when Object key starts with 'content'
Steps to reproduce:
	`let a = [{'name': 'description', 'content': 'This is the English META description'}];`
	`let b 	= [{'content': 'This is the English META description', 'name': 'description'}];`
1)	`<Helmet meta={a}/>
	Helmet.rewind().meta.toString() === '<meta data-react-helmet="true" name="description" content="This is the English META description"/>' //correct`
2)	`<Helmet meta={b}/>
	Helmet.rewind().meta.toString() === '' //incorrect`

Usually Object does not guarantee the key order, but:
	`console.log(1, a);`
	`console.log(2, b);`
Produces in browser (Chrome 47):
	`1 [ { content: 'This is the English META description', name: 'description' } ]`
	`2 [ { content: 'This is the English META description', name: 'description' } ]`
Produces in Node (0.12.7):
	`1 [ { name: 'description', content: 'This is the English META description' } ]`
	`2 [ { content: 'This is the English META description', name: 'description' } ]`

### PR DESCRIPTION
**Pull request by cwelch5**
Ordering of attributes doesn't matter when passing props to Helmet.  Special rule in place for when a canonical link tag uses 'href' and 'rel', where 'rel' will take priority.  Unit tests added to support.

Fixes #80

### CODE REVIEW

File: src/Helmet.js [hunk 1]
```diff
@@ -76,29 +76,38 @@ const getTagsFromPropsList = (tagName, validTags, propsList) => {
...
             instanceTags.filter(tag => {
+                let validAttributeKey;
                 for (const attributeKey of Object.keys(tag)) {
                     const lowerCaseAttributeKey = attributeKey.toLowerCase();
-                    const value = tag[attributeKey].toLowerCase();
 
-                    if (Object.is(validTags.indexOf(lowerCaseAttributeKey), -1)) {
-                        return false;
+                    // Special rule with link tags, since rel and href are both valid tags, rel takes priority
+                    if (validTags.indexOf(lowerCaseAttributeKey) > -1
```
Comment (doctyper) [inline src/Helmet.js]: You can use `!validTags.includes(lowerCaseAttributeKey)` here.

**Comment by potench** [comment]
+1 - good to merge.

**Comment by cwelch5** [comment]
@taak77 - it does include it, check out `index.js`

**Comment by taak77** [comment]
Yeah, i missed it in the diff.

**Pull request merged**